1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?
3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?
4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [1]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

def permutations(n):
    return int(factorial(n))

### Задание 1.
Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

a) В колоде 52 карты, 4 масти, значит каждой масти в колоде по 13 карт. Эта задача на сочетание.

Найдем сколькими способами сможем вытащить 4 карты из 13

In [12]:
C_13_4 = combinations(13, 4)
C_13_4

715

In [33]:
C_52_4 = combinations(52, 4)
C_52_4

270725

In [35]:
P = C_13_4 / C_52_4
P

0.0026410564225690276

Ответ: 0.0026410564225690276

б) В колоде всего 4 туза. Условие подразумевает, что тузов может быть от 1 до 4

##### Вариант 1

Вытаскиваем 1 туз и 3 не туза

In [64]:
C_4_1 = combinations(4, 1)
C_4_1

4

In [65]:
C_48_3 = combinations(48, 3)
C_48_3

17296

Вытаскиваем 2 туза и 2 не туза

In [66]:
C_4_2 = combinations(4, 2)
C_4_2

6

In [67]:
C_48_2 = combinations(48, 2)
C_48_2

1128

Вытаскиваем 3 туза и 1 не туз

In [68]:
C_4_3 = combinations(4, 3)
C_4_3

4

In [69]:
C_48_1 = combinations(48, 1)
C_48_1

48

Вытаскиваем 4 туза и 0 других карт

In [70]:
C_4_4 = combinations(4, 4)
C_4_4

1

Итоговая вероятность будет сумма всех 4 вариантов, а вероятность в варианте будет произведение вероятности вытащить туза и не туза

In [71]:
C = C_4_1 * C_48_3 + C_4_2 * C_48_2 + C_4_3 * C_48_1 + C_4_4
P = C / C_52_4
P

0.2812632745405855

Ответ: 0.2812632745405855

##### Вариант 2
Общая вероятность 1, вытащить хотя бы 1 туз будет вероятность 1 - вероятность не вытащить ни одного туза

In [52]:
C_48_4 = combinations(48, 4)
C_48_4

194580

In [54]:
P = 1 - C_48_4 / C_52_4
P

0.2812632745405854

### Задание 2
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Благоприятный вариант 1.
Общее число всех вариантов сочетание из 10 цифр выбрать 3

In [48]:
С_10_3 = combinations(10, 3)
C_10_3

120

In [50]:
P = 1 / C_10_3
P

0.008333333333333333

Ответ: 0.008333333333333333

### Задание 3
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Задача аналогична 1ой

In [28]:
C_9_3 = combinations(9, 3)
C_9_3

84

In [30]:
C_15_3 = combinations(15, 3)
C_15_3

455

In [31]:
P = C_9_3 / C_15_3
P

0.18461538461538463

Ответ: P = 0.18461538461538463

### Задание 4
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Вероятность вытащить 2 выигрышных билета

In [41]:
P_2 = permutations(2)
P_2

2

Вероятность вытащить любые 2 билета

In [46]:
A_100_2 = arrangements(100, 2)
A_100_2

9900

In [47]:
P = P_2 / A_100_2
P

0.00020202020202020202

Ответ: 0.00020202020202020202